packages/react/src/components/Dialog/index.tsx

```diff
@@ -53,24 +58,29 @@ export default class Dialog extends React.Component<DialogProps, DialogState> {
     this.close = this.close.bind(this);
     this.focusHeading = this.focusHeading.bind(this);
     this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.bindKeyboardEvents = this.bindKeyboardEvents.bind(this);
     this.state = {};
   }
 
   componentDidMount() {
     if (this.props.show) {
+      this.bindKeyboardEvents('added');
```

Comment:
Keeping in line with how we name these functions elsewhere (see https://github.com/dequelabs/cauldron/blob/256641b5883108529152d2a299088dcc6f7ef146/packages/react/src/components/ClickOutsideListener/index.tsx#L42) I would use `attachEventListeners` and `removeEventListeners` to make it a little more declarative. 

This is also in line with `attachIsolator`.